src/wallet/wallet.cpp

```diff
@@ -1753,7 +1753,10 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate, b
             }
 
             CBlock block;
-            ReadBlockFromDisk(block, pindex);
+            if (!ReadBlockFromDisk(block, pindex)) {
+                LogPrintf("Unable to read block %d (%s) from disk.", pindex->nHeight, pindex->GetBlockHash().ToString());
```

Comment:
would be good to add the function name in the logging as well, so we can distinguish it from other ones with the same text.
`LogPrintf("%s : Unable to...", __func__);`